keps/sig-scheduling/4664-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -369,15 +369,11 @@ when selecting victims.
 - if the priority of the preemptor is greater than or equal to the value of `AllowDisruptionByPriorityGreaterThanOrEqual` in victim pod, 
   the implementation will remain consistent with the existing behavior, meaning that the scheduler will try to select victims whose PDBs 
   are not violated by preemption, but if no such victims are found, preemption will still happen and lower priority pods will be preempted 
-  via the `/evictions` endpoint despite their PDBs being violated. If the `/eviction` endpoint returns a response `429 Too Many Requests`
-  and the scheduler will fallback to deletion as an alternative.
+  using the client-go's delete method.
```

Comment:
This part is confusing to me. Are we tring to introduce `/evictions` endpoint in this KEP ?

IIRC, we don't using eviction in preemption right now. https://github.com/kubernetes/kubernetes/blob/4a668bcf143c0652882d2e59051dcf0cd843c24c/pkg/scheduler/util/utils.go#L137

Besides, I don't think we should use `/evictions` endpoint here.
During the preemption, we'll separate victims into two parts: nonViolatingCandidates, violatingCandidates.

Since we already know in advance which pods' deletion would violate the PDB.

1.  Using `eviction` on nonViolatingCandidates will equal to delete it directly(?)
2. Using `eviction` on violatingCandidates will always return 429. And since we have already deleted all nonViolatingCandidates  on the same node before trying to delete violatingCandidates, this will results in all violatingCandidates  can't be deleted and thus preemption will always fail when deleting all nonViolatingCandidates is not enough.

Besides, if we use `/evictions` endpoint for all pods in preemption. Then what's the point of this KEP? We'll deny any preempter if pdb is violated.
Also, use `/evictions` endpoint for all pods would also be a breaking change and is not backward compatible.
@alculquicondor @Huang-Wei

Reply:
Please take a look at the [original issue](https://github.com/kubernetes/kubernetes/issues/91492) to understand the drive behind this KEP.

Also take a look at this [comment](https://github.com/kubernetes/enhancements/pull/3755#discussion_r1089155881) to understand the reason behind the eviction first policy.

Reply:
Thanks! I read the original issue and comment you left in #3280

The drive behind the KEP and my understanding while reading it seem to be different. I thought the KEP was introducing a new field to make it harder for specific pods to be preempted during scheduling. However, it appears that the original intention is for the scheduler to respect PDBs in every preemptive case.

In this case, I agree that we should use eviction api in the whole preemption.